lib/rspecq/worker.rb

```diff
@@ -69,6 +76,8 @@ def work
         # to `requeue_lost_job` inside the work loop
         update_heartbeat
 
+        break if @queue.suite_failed?
```

Comment:
Can't this be moved with the other exit cases? (line 89):

```ruby
        # build is finished
        return if (job.nil? && queue.exhausted?) || @queue.suite_failed?
```

Reply:
I do not think it should, since before this check, the worker requeues a failed job and pops something from the queue